### PR TITLE
refactor: useFacet into useSearch

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
Using useSearch instead of useFacet.

## Description
VSF core team created useSearchFactory as a better solution than using useFacetFactory. In this pr useFacet has been changed into useSearch.

### CODE REVIEW

**Review by scernisevs — COMMENTED** [review]
hi @WojtekTheWebDev ,

I think we would need to also update the references in
theme/components/CategoryPageHeader.vue,
theme/components/FiltersSidebar.vue

could you please have a look at those 2 files as well? :) 
otherwise the PLP might brake.

File: packages/theme/pages/Category.vue [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+  middleware: cacheControl({
+    'max-age': 60,
+    'stale-when-revalidate': 5
+  }),
+  setup(props, context) {
+    const th = useUiHelpers();
+    const uiState = useUiState();
+    const { addItem: addItemToCart, isInCart } = useCart();
+    const { addItem: addItemToWishlist, isInWishlist, removeItem: removeItemFromWishlist } = useWishlist();
+    const { result, search, loading, error } = useSearch();
```
Comment (WojtekTheWebDev) [inline packages/theme/pages/Category.vue:233]: Here is the change ;)

**Comment by WojtekTheWebDev** [comment]
> hi @WojtekTheWebDev ,
> 
> Thank you very much for raising a PR for this.
> 
> I wonder if we would need to also update the references in theme/components/CategoryPageHeader.vue, theme/components/FiltersSidebar.vue ?
> 
> could you please have a look at those 2 files as well? :)
> 
> Thank you!

Thanks, will do that!

**Comment by WojtekTheWebDev** [comment]
> hi @WojtekTheWebDev ,
> 
> Thank you very much for raising a PR for this.
> 
> I wonder if we would need to also update the references in theme/components/CategoryPageHeader.vue, theme/components/FiltersSidebar.vue ?
> 
> could you please have a look at those 2 files as well? :)
> 
> Thank you!

Done

**Review by scernisevs — APPROVED** [review]
LGTM

**Pull request merged**